platform/linuxbsd/freedesktop_portal_desktop.cpp

```diff
@@ -617,6 +749,263 @@ void FreeDesktopPortalDesktop::_system_theme_changed_callback() {
 	}
 }
 
+void FreeDesktopPortalDesktop::_status_notifier_item_unregister(DBusConnection *connection, void *user_data) {
+}
+
+DBusHandlerResult FreeDesktopPortalDesktop::_status_notifier_item_handle_message(DBusConnection *connection, DBusMessage *message, void *user_data) {
+	FreeDesktopPortalDesktop *portal = (FreeDesktopPortalDesktop *)user_data;
+	ERR_FAIL_NULL_V(portal, DBUS_HANDLER_RESULT_HANDLED);
+
+	String interface = dbus_message_get_interface(message);
+	String member = dbus_message_get_member(message);
+
+	const char *destination_ptr = dbus_message_get_destination(message);
+	ERR_FAIL_NULL_V(destination_ptr, DBUS_HANDLER_RESULT_HANDLED);
+
+	String destination;
+	destination.parse_utf8(destination_ptr);
+
+	ERR_FAIL_COND_V(!portal->indicator_id_map.has(destination), DBUS_HANDLER_RESULT_HANDLED);
+	DisplayServer::IndicatorID id = portal->indicator_id_map[destination];
+
+	ERR_FAIL_COND_V(!portal->indicators.has(id), DBUS_HANDLER_RESULT_HANDLED);
+	StatusNotifierItem &item = portal->indicators[id];
+
+	if (dbus_message_is_method_call(message, BUS_INTERFACE_STATUS_NOTIFIER_ITEM, "Activate")) {
+		if (item.activate_callback.is_valid()) {
+			item.activate_callback.call_deferred();
+		}
+
+		return DBUS_HANDLER_RESULT_HANDLED;
+	}
+
+	if (dbus_message_get_type(message) == DBUS_MESSAGE_TYPE_METHOD_CALL && interface == BUS_INTERFACE_PROPERTIES && member == "Get") {
+		const char *object_name;
+		const char *property;
+
+		DBusError error;
+		dbus_error_init(&error);
+		dbus_message_get_args(message, &error, DBUS_TYPE_STRING, &object_name, DBUS_TYPE_STRING, &property);
+
+		dbus_bool_t success = FALSE;
+
+		if (strcmp(object_name, "org.freedesktop.StatusNotifierItem") == 0) {
+			DBusMessage *reply = dbus_message_new_method_return(message);
+
+			// TODO: Error handling.
+			DBusMessageIter reply_iter;
+			dbus_message_iter_init_append(reply, &reply_iter);
+
+			if (strcmp(property, "Category") == 0) {
+				success = _dbus_messsage_iter_append_basic_variant(&reply_iter, DBUS_TYPE_STRING, "ApplicationStatus");
+			} else if (strcmp(property, "Id") == 0) {
+				// TODO: No idea what would be a good value.
+				// The documentation says that this is pretty much an app id.
+				success = _dbus_messsage_iter_append_basic_variant(&reply_iter, DBUS_TYPE_STRING, "org.godotengine.Godot");
+			} else if (strcmp(property, "Title") == 0) {
+				// TODO: No idea what would be a good value.
+				// Same thing as above, although it's meant to be a little bit more verbose.
+				success = _dbus_messsage_iter_append_basic_variant(&reply_iter, DBUS_TYPE_STRING, "Godot game engine");
```

Comment:
I guess it should be something like this:
https://github.com/godotengine/godot/blob/29b3d9e9e538f0aa8effc8ad8bf19a2915292a89/platform/linuxbsd/tts_linux.cpp#L59-L65

ID can be generated from the same project settings (it's done in multiple places for other platforms), but I feel like we might want it as a dedicated project setting, since unlike title, IDs are usually limited to basic Latin, numbers and a few separator characters.